tests/zfs-tests/tests/functional/cli_root/zfs_unmount/zfs_unmount_all_002_pos.ksh

```diff
@@ -0,0 +1,203 @@
+#
+
+verify_runnable "both"
+
+set -A fs "$TESTFS" "$TESTFS1"
+set -A ctr "" "$TESTCTR" "$TESTCTR1" "$TESTCTR/$TESTCTR1"
+set -A vol "$TESTVOL" "$TESTVOL1"
+
+# Test the mounted state of root dataset (testpool/testctr)
+typeset mnt=$TESTCTR
+
+function setup_all
+{
+	typeset -i i=0
+	typeset -i j=0
+	typeset path
+
+	while (( i < ${#ctr[*]} )); do
+
+		path=${TEST_BASE_DIR%%/}/testroot$$/$TESTPOOL
+		if [[ -n ${ctr[i]} ]]; then
+			path=$path/${ctr[i]}
+
+			setup_filesystem "$DISKS" "$TESTPOOL" \
+				"${ctr[i]}" "$path" \
+				"ctr"
+		fi
+
+		if is_global_zone ; then
+			j=0
+			while (( j < ${#vol[*]} )); do
+				setup_filesystem "$DISKS" "$TESTPOOL" \
+				"${ctr[i]}/${vol[j]}" \
+					"$path/${vol[j]}" \
+					"vol"
+				((j = j + 1))
+			done
+		fi
+		j=0
+		while (( j < ${#fs[*]} )); do
+			setup_filesystem "$DISKS" "$TESTPOOL" \
+				"${ctr[i]}/${fs[j]}" \
+				"$path/${fs[j]}"
+			((j = j + 1))
+		done
+
+		((i = i + 1))
+	done
+
+	return 0
+}
+
+function cleanup_all
+{
+	typeset -i i=0
+	typeset -i j=0
+
+	((i = ${#ctr[*]} - 1))
+
+	while (( i >= 0 )); do
+		if is_global_zone ; then
+			j=0
+			while (( j < ${#vol[*]} )); do
+				cleanup_filesystem "$TESTPOOL" \
+					"${ctr[i]}/${vol[j]}"
+				((j = j + 1))
+			done
+		fi
+
+		j=0
+		while (( j < ${#fs[*]} )); do
+			cleanup_filesystem "$TESTPOOL" \
+				"${ctr[i]}/${fs[j]}"
+			((j = j + 1))
+		done
+
+		[[ -n ${ctr[i]} ]] && \
+			cleanup_filesystem "$TESTPOOL" "${ctr[i]}"
+
+		((i = i - 1))
+	done
+
+	[[ -d ${TEST_BASE_DIR%%/}/testroot$$ ]] && \
+		rm -rf ${TEST_BASE_DIR%%/}/testroot$$
+}
+
+#
+# This function verifies that the file systems in $mnt are unmounted.
+# Next, it ensures that all other file systems in remain mounted.
+#
+function verify_related
+{
+	typeset -i i=0
+	typeset -i j=0
+	typeset path
+
+	while (( i < ${#ctr[*]} )); do
+
+		if { [[ ${ctr[i]} == $mnt ]] || [[ ${ctr[i]} == $mnt/* ]] }; then
+			logfunc=log_must
+		else
+			logfunc=log_mustnot
+		fi
+
+		path=$TESTPOOL
+		[[ -n ${ctr[i]} ]] && \
+			path=$path/${ctr[i]}
+
+		if is_global_zone ; then
+			j=0
+			while (( j < ${#vol[*]} )); do
+				log_must unmounted "$path/${vol[j]}"
+				((j = j + 1))
+			done
+		fi
+
+		j=0
+		while (( j < ${#fs[*]} )); do
+			$logfunc unmounted "$path/${fs[j]}"
+			((j = j + 1))
+		done
+
+		$logfunc unmounted "$path"
+
+		((i = i + 1))
+	done
+
+	return 0
+}
+
+
+log_assert "Verify that 'zfs $unmountall filesystem' succeeds as root, " \
+	"and all available related ZFS filesystems are unmounted."
+
+log_onexit cleanup_all
+
+log_must setup_all
+
+typeset opt
+for opt in "-a" "-fa"; do
+	export __ZFS_POOL_RESTRICT="$TESTPOOL"
+	log_must zfs $mountall
+	unset __ZFS_POOL_RESTRICT
+
+	export __ZFS_POOL_RESTRICT="$TESTPOOL"
+	log_must zfs unmount $opt $TESTPOOL/$mnt
+	unset __ZFS_POOL_RESTRICT
+
+	log_must verify_related
+	log_note "Verify that 'zfs $mountcmd' will display " \
+	"all ZFS filesystems related to '$TESTPOOL/$mnt' are currently unmounted."
```

Comment:
`all ZFS filesystems related to` should be `all available ZFS filesystems related to`